### ISSUE TITLE
[dagit] reexecution doesn't respect solid subset selection

### ISSUE THREAD

**Issue by alangenfeld** [dagster-io/dagster#1223]
> Taylor [2:33 PM]
> I think I found a bug in single step execution
> str:
> 1. select a subset of a dag and execute it
> 2. when a solid in that subset fails, execute only that solid
> 3. note that you are taken to a view of the entire pipeline, not the subset. That's jarring and hard to follow
> 
> incidentally the re-execution did work

**Comment by bengotow** [comment]
Will investigate this next. Seems like a definite bug.
